src/site/topics/queues/linked-queue.rst

```diff
@@ -2,270 +2,226 @@
-.. code-block:: java
-    :linenos:
-
-    @Override
-    public T dequeue() {
-        if (isEmpty()) {
-            throw new NoSuchElementException("Dequeueing from an empty queue.");
-        }
-        T returnElement = front.getData();
-        front = front.getNext();
-        size--;
-        if (isEmpty()) {
-            rear = null;
-        }
-        return returnElement;
-    }
-
-    @Override
-    public T first() {
-        if (isEmpty()) {
-            throw new NoSuchElementException("First from an empty queue.");
-        }
-        return front.getData();
-    }
+
+
+``dequeue`` & ``first``
+-----------------------
+
+.. literalinclude:: /../main/java/LinkedQueue.java
+    :language: java
+    :lineno-match:
+    :lines: 42-63
+
+
 
 * Like ``LinkedStack`` and ``ArrayStack``, trying to access something from the empty queue throws an exception
+* ``dequeue`` does a *remove/delete from the front of a linked structure*
 
-* Notice in ``dequeue`` that, if it's not empty, we just *remove/delete from the front of a linked structure*
-    * This was the same as ``pop`` in the ``LinkedStack``
+* Notice the edge case of a ``dequeue`` resulting in an empty queue
 
-* In addition to being empty, the only other edge case we need to watch out for is if the ``dequeue`` makes the queue empty
-* If this happens, we must set ``front`` to ``null``
-    * This is actually taken care of already since ``front``'s next would be ``null``, and saying ``front = front.getNext()`` makes ``front`` ``null``
+    * The update to the ``front`` will not be a problem
 
-* We should also set ``rear`` to ``null``
-* This helps with garbage collection and keeping the state of the queue *correct*
-* If we don't do this, ``rear`` will continue to point to a node that should not be in the queue anymore
-    * Our current implementation will handle this scenario fine as an enqueue on an empty queue sets both ``front`` and ``rear`` to ``null``
-    * But imagine also having a messed up count and ``enqueuing`` after this without setting ``rear`` to ``null``
-    * If we ``enqueue`` in this case, we might end up saying ``rear.setNext(toEnqueue)``
-    * What would that mean?
-    * What would that look like?
+        * ``front = front.getNext()`` will set ``front`` to ``null`` since ``front.getNext()`` in this situation would return ``null``
 
+    * The trouble is with ``rear`` --- typically ``rear`` is left alone with a ``dequeue``
+    * However, if the ``dequeue`` resulted in an empty queue, it would be left referencing the ``Node`` that was just removed
+    * The solution is to simply set it to ``null`` if the stack is empty after the dequeue
 
-* What is the computational complexity of a ``dequeue``?
 
-Variations
-==========
+.. note::
+
+    With the current implementation of the ``LinkedQueue``, missing this edge case would not actually cause a problem
+    with the functionality of the queue. However, it is still useful to remove any *dangling* edge cases.
```

Comment:
It'd be a problem in general. Enqueuing is `rear = rear.next = new Node(..)`. Dequeuing afterwards is still trying to remove from an empty list, as `front` is still `null`.

You'd have to write empty checks is such a way to account for this edge case to not be logically relevant.

Reply:
I'm not clear on what you mean. With the current implementation, it shouldn't be a problem as `enqueue` checks the edge case. 

```java
    public boolean enqueue(T element) {
        Node<T> toEnqueue = new Node<>(element);
        if (isEmpty()) {
            front = toEnqueue;
        } else {
            rear.setNext(toEnqueue);
        }
        rear = toEnqueue;
        size++;
        return true;
    }
```

Reply:
Was your point that this edge case being caught in the `enqueue` would be good to emph in the note?

Reply:
> With the current implementation, it shouldn't be a problem as `enqueue` checks the edge case.

Your code explicitly handles this edge case, but the note mentions you can ignore the edge case. These are contradictory.